scripts/globals/items/dusty_elixir.lua

```diff
@@ -0,0 +1,53 @@
+-----------------------------------------
+-- ID: 5433
+-- Item: Dusty Elixir
+-- Item Effect: Instantly restores 25% of HP and MP
+-----------------------------------------
+require("scripts/globals/status")
+require("scripts/globals/msg")
+-----------------------------------------
+-- OnItemCheck
+-----------------------------------------
+
+function onItemCheck(target)
+    local result = 0
+    if target:getMaxHP() == target:getHP() then
+        result = 56 -- Does not let player use item if their hp is full
+    end
+    if target:hasStatusEffect(dsp.effect.MEDICINE) then
+        result = 111
+    end
+    return result
+end
+
+-----------------------------------------
+-- OnItemUse
+-----------------------------------------
+
+function onItemUse(target)
+	local hpAmount  = 0
+    local mpAmount  = 0
+    local maxHP     = target:getMaxHP()
+    local maxMP     = target:getMaxMP()
+    local currentHP = target:getHP()
+    local currentMP = target:getMP()
+    local recoverHP = (maxHP/100)*25
+    local recoverMP = (maxMP/100)*25
+    local hpDiff    = maxHP - currentHP
+    local mpDiff    = maxMP - currentMP
+
```

Comment:
we do in fact have functions for percentages of hp or mp. you don't need to script the math.

Reply:
Looked into a simpler way and changed as suggested